### PR TITLE
Set species to model

### PR DESCRIPTION
**Pull request by lvnhmd**

### CODE REVIEW

File: pages/pil/update/index.js [hunk 1]
```diff
@@ -33,10 +35,17 @@ module.exports = settings => {
...
+
+    const trainingSpecies = flattenDeep(req.model.certificates.map(c => c.modules.map(m => m.species)));
+    const exemptionsSpecies = flattenDeep(req.model.exemptions.map(e => e.species));
+    const species = trainingSpecies.concat(exemptionsSpecies);
+
     req.model = {
       ...req.pil,
-      ...values
+      ...values,
+      species: species.filter((v, i) => species.indexOf(v) === i)
```
Comment (lennym) [inline pages/pil/update/index.js]: Is this effectively a uniqueness test? If so could we use something a bit more explicit? i.e. https://lodash.com/docs/4.17.11#uniq

File: pages/pil/update/index.js [hunk 2]
```diff
@@ -8,6 +8,8 @@ const procedures = require('../procedures');
 const exemptions = require('../exemptions');
 const training = require('../training');
 
+const flattenDeep = (arr) => { return arr.reduce((acc, val) => Array.isArray(val) ? acc.concat(flattenDeep(val)) : acc.concat(val), []); };
```
Comment (joefitter) [inline pages/pil/update/index.js]: https://lodash.com/docs/4.17.11#flattenDeep

File: pages/pil/update/index.js (hunk 1 above)
Comment (lvnhmd) [reply pages/pil/update/index.js]: done

File: pages/pil/update/index.js (hunk 2 above)
Comment (lvnhmd) [reply pages/pil/update/index.js]: done

**Pull request merged**